### ISSUE TITLE
Canvas color input

### ISSUE THREAD

**Issue by wstaelens** [synfig/synfig#637]
**Synfig version & platform**:
Synfig 1.2.2 - Windows 10 x64 Pro

**Issue description**:

There is no input restriction for color when setting color for a rectangle.
it should not be allowed to input #ffffffffffffff as a correct hex value. Text should stop at either 6 'F'-characters or 8 'F'-chars (when using alpha).

![synfig color input](https://user-images.githubusercontent.com/7447285/46342355-cf1d3200-c63a-11e8-912c-fd3be42d5bc1.PNG)

The label "Color name:" should also be better changed to "HEX value:"

**Comment by thekalinga** [comment]
A simple regex validation should be good enough

**Comment by morevnaproject** [comment]
Hey, @SharmaRithik! Any progress on this? ^__^

**Comment by SharmaRithik** [comment]
finding a solution for it, will come up with a pull request soon.
Got the file and read your comment too. :-)

**Comment by morevnaproject** [comment]
@SharmaRithik If you need any help - just let me know. ^__^

**Comment by SharmaRithik** [comment]
The range is  between #000000 - #FFFFFF so we could restrict inputs, that
values cannot be entered out of this range.

**Comment by wstaelens** [comment]
Well HEX colors can use:

- 6 digits (#000000) (#RRGGBB)
- 8 digits (#000000FF) (#RRGGBBAA)
- 3 digits (#000) (#RGB)
- 4 digits (#000F) (#RGBA)

CSS4 allows these four notations as valid HEX notations:
https://drafts.csswg.org/css-color/#hex-notation

I suggest you also accept this.

**Comment by BobSynfig** [comment]
In fact the way this input works has been done on purpose, featuring _gdk_rgba_parse_ but its label is not very descriptive.

From https://developer.gnome.org/gdk3/stable/gdk3-RGBA-Colors.html#gdk-rgba-parse

> **_gboolean gdk_rgba_parse (GdkRGBA *rgba, const gchar *spec);_**
>
> Parses a textual representation of a color, filling in the red , green , blue and alpha fields of the rgba GdkRGBA.
>
> The string can be either one of:
> 
> * A **standard name** (Taken from the [X11 rgb.txt file](https://en.wikipedia.org/wiki/X11_color_names)).
> * A hexadecimal value in the form “**#rgb**”, “**#rrggbb**”, “**#rrrgggbbb**” or ”**#rrrrggggbbbb**”
> * A RGB color in the form “**rgb(r,g,b)**” (In this case the color will have full opacity)
> * A RGBA color in the form “**rgba(r,g,b,a)**”
> 
> Where “r”, “g”, “b” and “a” are respectively the red, green, blue and alpha color values. In the last two cases, r g and b are either integers in the range 0 to 255 or percentage values in the range 0% to 100%, and a is a floating point value in the range 0 to 1.

**Comment by ankit-kumar-dwivedi** [comment]
So is this still an issue? as it's clear there is a purpose why the input field is designed this way. Although we can restrict it to accept only 8 digit hexcode that includes every color with alpha.

**Comment by ankit-kumar-dwivedi** [comment]
> Personnaly I think it would be better just to add an explanation, for example with a button with a question mark displaying a help window.
> Why to restrict a very powerful feature?

Yes @BobSynfig I totally agree to this. @morevnaproject can you change the topic of this issue :-)
>In this case I suggest to rename "Color Name" to "HTML code".

I think we should keep it as is or change it to just "color" with an ℹ️ tag

**Comment by morevnaproject** [comment]
>> In this case I suggest to rename "Color Name" to "HTML code".

> I think we should keep it as is or change it to just "color" with an :information_source: tag

Still, I would insist on "HTML code". ^__^

Quick overview what's going on in other graphic editors:

**Blender**

![screenshot_004](https://user-images.githubusercontent.com/332868/51802758-a0793600-227f-11e9-9ac8-d8f9384d7bc9.png)

* Filed name: "HEX"
* Behavior: Allows to enter any symbols, string length limited to 7. Any value not corresponding to hex code resets color to black.

**Gimp**

![screenshot_005](https://user-images.githubusercontent.com/332868/51802793-e59d6800-227f-11e9-9553-e7fde6a0f339.png)

* Filed name: "HTML notation"
* Behavior: Allows to enter any symbols, string length not limited. Accepts hex and color names (black, white, green, etc). Entering incorrect value resets color to black.

**Krita**

![screenshot_006](https://user-images.githubusercontent.com/332868/51802817-2bf2c700-2280-11e9-98a9-12130bd1ec77.png)

* Filed name: "Color name"
* Behavior: Allows to enter any digits and letters [a,b,c,d,e,f,#]. String length is limited to 7 if stats with "#", otherwise limited to 6 ("#" added automatically if missing). Accepts hex codes only. Entering incorrect value technically not possible.

**Comment by morevnaproject** [comment]
On fact, I prefer to drop the possibility to provide color names - I never use it by myself and in general artists prefer to avoid "clean" colors (with exception of "white" and "black"). Also, there are only 16 colors named in HTML, so this is not a big loss.

In return, we get a simple way to fix this issue - I like Krita's or Blender's approach. ^__^

**Comment by wstaelens** [comment]
I prefer Blender's way

**Comment by morevnaproject** [comment]
> In fact a bit more than 16 ;) https://en.wikipedia.org/wiki/X11_color_names#Color_name_chart

Oh, I was looking at the wrong page (https://en.wikipedia.org/wiki/Web_colors) ^__^"
You right, all colors from list of "X11 color names" are supported currently by Synfig's text entry.

**Comment by bajajvinamr** [comment]
Is this issue open?

**Comment by 002harshit** [comment]
can i get assigned if it's still open?

**Comment by BobSynfig** [comment]
As long as we can still use X11 color names and not introducing regressions in functionnalities ...
The control should be limited to values starting by a # (read carefully all the comments)
It's easier to type "lavender" than remember #E6E6FA
